sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/AadResourceServerWebSecurityConfigurerAdapter.java

```diff
@@ -5,39 +5,60 @@
 
 import com.azure.spring.cloud.autoconfigure.aad.implementation.jwt.AadJwtGrantedAuthoritiesConverter;
 import com.azure.spring.cloud.autoconfigure.aad.properties.AadResourceServerProperties;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.core.convert.converter.Converter;
 import org.springframework.security.authentication.AbstractAuthenticationToken;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.annotation.web.configurers.oauth2.server.resource.OAuth2ResourceServerConfigurer;
 import org.springframework.security.oauth2.jwt.Jwt;
 import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationConverter;
+import org.springframework.security.web.SecurityFilterChain;
 import org.springframework.util.StringUtils;
 
 /**
- * Abstract configuration class, used to make JwtConfigurer and AADJwtBearerTokenAuthenticationConverter take effect.
+ * Abstract configuration class, used to make {@link OAuth2ResourceServerConfigurer.JwtConfigurer } and {@link AadJwtGrantedAuthoritiesConverter} take effect.
  *
- * @see WebSecurityConfigurerAdapter
  */
-public abstract class AadResourceServerWebSecurityConfigurerAdapter extends WebSecurityConfigurerAdapter {
+public abstract class AadResourceServerWebSecurityConfigurerAdapter {
```

Comment:
Could you help explain this change?

Reply:
The `WebSecurityConfigurerAdapter ` has already been deprecated, current usage will be recommended, see more from https://spring.io/blog/2022/02/21/spring-security-without-the-websecurityconfigureradapter

Reply:
The `WebSecurityConfigurerAdapter` was deprecated and the Security encourages users to move towards a component-based security configuration.
We should not keep the adapter too, but another way should be provided for Azure AD users, a HTTP security configurer should be available for users to use directly, it will be a variant of the current `AadResourceServerWebSecurityConfigurerAdapter`, see more https://docs.spring.io/spring-security/reference/6.0.0-M7true/servlet/configuration/java.html#jc-custom-dsls

Reply:
Follow-ups: 
- What's the difference between WebSecurity and HttpSecurity?
- What's the default path if we configure the HttpSecurity this way?

Reply:
What's the difference between WebSecurity and HttpSecurity?
 -> A good blog for WebSecurity vs HttpSecurity https://ravthiru.medium.com/springboot-security-configuration-using-httpsecurity-vs-websecurity-1a7ec6a23273
More code logic can be found here: https://github.com/spring-projects/spring-security/blob/main/config/src/main/java/org/springframework/security/config/annotation/web/builders/WebSecurity.java#L277-L330


What's the default path if we configure the HttpSecurity this way?
 -> We do not configure any path filter by default, it will be configured by customers.